src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeSettingsFilter.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.cloud.azure.management;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.cloud.azure.AzureSettingsFilter;
+import org.elasticsearch.common.component.AbstractLifecycleComponent;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.settings.SettingsFilter;
+
+import static org.elasticsearch.cloud.azure.management.AzureComputeService.Management.*;
+
+public class AzureComputeSettingsFilter extends AbstractLifecycleComponent<AzureSettingsFilter>
+    implements AzureSettingsFilter {
```

Comment:
I think the `AzureSettingsFilter` interface is not needed - you can just bind concrete implementations. There is also no need for this to be a lifecycle component since lifecycle events are not used anyway.